gazebo_ros_control/src/default_robot_hw_sim.cpp

```diff
@@ -331,11 +331,19 @@ void DefaultRobotHWSim::writeSim(ros::Time time, ros::Duration period)
           switch (joint_types_[j])
           {
             case urdf::Joint::REVOLUTE:
-              angles::shortest_angular_distance_with_limits(joint_position_[j],
-                                                            joint_position_command_[j],
-                                                            joint_lower_limits_[j],
-                                                            joint_upper_limits_[j],
-                                                            error);
+              if(joint_upper_limits_[j] - joint_lower_limits_[j] > 2.0 * M_PI)
```

Comment:
Use absolute value to determine if the limits (the rotatable revolutions) exceeds 2*pi

Reply:
Do you mean `joint_upper_limits_[j]` can be smaller than `joint_lower_limits_[j]` ?

Reply:
oops i forgot about the upper > lower limit
this should be ok